spring-cloud-kubernetes-fabric8-discovery/src/main/java/org/springframework/cloud/kubernetes/fabric8/discovery/KubernetesDiscoveryClient.java

```diff
@@ -135,52 +130,47 @@ else if (properties.namespaces().isEmpty()) {
 	}
 
 	private List<ServiceInstance> getNamespaceServiceInstances(EndpointSubsetNS es, String serviceId) {
-		String namespace = es.namespace();
+
 		List<EndpointSubset> subsets = es.endpointSubset();
+		if (subsets.isEmpty()) {
+			LOG.debug(() -> "serviceId : " + serviceId + " does not have any subsets");
+			return List.of();
+		}
+
+		String namespace = es.namespace();
 		List<ServiceInstance> instances = new ArrayList<>();
-		if (!subsets.isEmpty()) {
-			Service service = client.services().inNamespace(namespace).withName(serviceId).get();
-			Map<String, String> serviceMetadata = serviceMetadata(serviceId, service, properties);
-			KubernetesDiscoveryProperties.Metadata metadataProps = properties.metadata();
-
-			for (EndpointSubset s : subsets) {
-				// Extend the service metadata map with per-endpoint port information (if
-				// requested)
-				Map<String, String> endpointMetadata = new HashMap<>(serviceMetadata);
```

Comment:
port metadata has moved to the method that does all the service's metadata computation.